app/assets/javascripts/editor.js

```diff
@@ -280,6 +279,39 @@ function (ko, models, tools, msg, validate, owned) {
             }
         };
 
+        self.doUpdateFigureCaption = function () {
```

Comment:
You are actually uploading the whole figure object, therefore I would call it `doUpadteFigure`